lib/babe/epoch_handler.go

```diff
@@ -75,83 +75,27 @@ func (h *epochHandler) run(ctx context.Context, errCh chan<- error) {
 		return
 	}
 
-	// for each slot we're handling, create a timer that will fire when it starts
-	// we create timers only for slots where we're authoring
-	authoringSlots := getAuthoringSlots(h.slotToPreRuntimeDigest)
+	logger.Debugf("authoring in %d slots in epoch %d", len(h.slotToPreRuntimeDigest), h.epochNumber)
 
-	type slotWithTimer struct {
-		startTime time.Time
-		timer     *time.Timer
-		slotNum   uint64
-	}
+	for {
+		currentSlot, err := h.slotHandler.waitForNextSlot(ctx)
+		if errors.Is(err, context.DeadlineExceeded) {
+			errCh <- nil
+			return
```

Comment:
just to to clarify, you don't want to continue to call `handleSlot` if there is a timeout?

Reply:
mybad I changed it to be any error:

```
currentSlot, err := h.slotHandler.waitForNextSlot(ctx)
if err != nil {
  errCh <- err
  return
}
```